### PR TITLE
Add user-generated annotations tests to the server

### PR DESCRIPTION
**Pull request by mweiden**
Partially completes https://github.com/chanzuckerberg/cellxgene/issues/969

This PR adds the following test cases (annotations API endpoints):
- make sure that the stuff saved in the file is what you expect, was everything saved (pytest)
- checks that you load the file correctly on startup (pytest)

In addition, all test data is moved to the test_datasets folder.

It does not cover the remaining test cases from https://github.com/chanzuckerberg/cellxgene/issues/969:
- all the above tests, but throw in some undo redo, (add a category, undo, then redo)

Note the undo/redo test is blocked by #1153

### CODE REVIEW

**Comment by mweiden** [comment]
There are some (potential) issues blocking this PR from being merged:
1. Annotations files generated with runs of cellxgene with a `cxg` file (`cellxgene launch <options> <data_file>.cxg`) don't seem to reload properly on subsequent runs with the `--experimental-annotations-file` flag
2. When `cellxgene launch` is run with a `cxg` file, new observations sent to the API via `PUT annotations/obs` cannot be recovered from the API via `GET annotations/obs`.

**Comment by mweiden** [comment]
@bkmartinjr @bmccandless There were a lot of files that needed to be formatted to pass `flake8`. Reviewing this will be easier if you just look at 00c6eb7.

**Comment by bkmartinjr** [comment]
> @bkmartinjr @bmccandless There were a lot of files that needed to be formatted to pass `flake8`. Reviewing this will be easier if you just look at [00c6eb7](https://github.com/chanzuckerberg/cellxgene/commit/00c6eb7c29fbfcb2eeb16baf46eda0445be0c897).


I thought the CI tests would reject anything with lint.  Not true?

File: example-dataset/pbmc3k-annotations-cxg.csv [hunk 1]
```diff
@@ -0,0 +1,2641 @@
+# Annotations generated on 2020-02-22T12:26:07 using cellxgene version 0.14.0
```
Comment (bkmartinjr) [inline example-dataset/pbmc3k-annotations-cxg.csv]: If this file is solely for testing (which I think it is), it should go in server/tests/test_datasets

I think same is true for the `.cxg` file in this directory

Comment (mweiden) [reply example-dataset/pbmc3k-annotations-cxg.csv]: Oh, pardon. I meant to remove this file. I can move the other files.

**Pull request merged**